### PR TITLE
Iteration 5

### PR DESCRIPTION
**Pull request by RajaaBoulassouak**
Rajaa Wrote Tests
Rajaa Implemented 
____ Reviewed
# Neccesary checkmarks:
- [x] All Tests are Passing

- [x] The code will run locally

## Type of change
- [x] New feature 
- [] Bug Fix

# Implements/Fixes:
- created RUD for invoices 
- built merchant-invoice relationship
closes #

# Check the correct boxes

- [x] This broke nothing
- [] This broke some stuff
- [] This broke everything

# Testing Changes
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

# Please Include a link to a gif of your feelings about this branch
Link:

### CODE REVIEW

File: app/controllers/little_shop_app.rb [hunk 1]
```diff
@@ -66,4 +67,29 @@ class LittleShopApp < Sinatra::Base
...
     erb :'items/show'
   end
+  
+  get '/invoices' do
+    @invoices = Invoice.all
+    erb :'invoices/index'
+  end
+  
+  get '/invoices/:id/edit' do
+    @invoice = Invoice.find_by( id: params[:id] )
```
Comment (patrickshobe) [inline app/controllers/little_shop_app.rb]: @RajaaBoulassouak You can use find instead of find_by for this, same with line 87

```ruby
@invoice = Invoice.find(params[:id])
```

File: db/schema.rb [hunk 2]
```diff
@@ -24,10 +32,11 @@
...
   end
 
-  create_table "merchants", force: :cascade do |t|
+  create_table "merchants", id: false, force: :cascade do |t|
+    t.integer "id"
     t.text "name"
-    t.datetime "created_at", null: false
-    t.datetime "updated_at", null: false
+    t.date "created_at"
+    t.date "updated_at"
```
Comment (patrickshobe) [inline db/schema.rb]: @RajaaBoulassouak Why did you change this to a date instead of a datetime?

``

Comment (RajaaBoulassouak) [reply db/schema.rb]: I didn't touch it, the schema got created like that when migrated. I entered:
"t.timestamps null: false" 
in the invoice migrate file like we did for merchant. Also observed that file has:
"class CreateInvoices < ActiveRecord::Migration[5.2]" 
instead of class 
"class CreateInvoicesTable < ActiveRecord::Migration[5.2]" 
I don't know why.
